packages/common/lib/common/tools.js

```diff
@@ -3439,9 +3439,36 @@ async function getInstancesOrderedByStartPrio(objects, logger, logPrefix = '') {
  * @returns {Promise<void>}
  */
 async function setExecutableCapabilities(execPath, capabilities, modeEffective, modePermitted, modeInherited) {
+    if (!Array.isArray(capabilities) || !capabilities.length) {
+        throw new Error('No capabilities array provided');
+    }
+
     // if not linux do nothing and silently exit
     if (os.platform() === 'linux') {
```

Comment:
Since this function is getting long, I'd turn this check around: 
```suggestion
    if (os.platform() !== 'linux') return;
```

Reply:
+1 as mentioned https://github.com/ioBroker/ioBroker.js-controller/pull/1756#discussion_r800234052

Reply:
Ok I do in my fork